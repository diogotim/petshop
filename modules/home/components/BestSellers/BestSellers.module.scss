.root h2 {
    font-weight: 900;
    text-align: center;
}