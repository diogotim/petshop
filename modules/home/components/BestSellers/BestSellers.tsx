import React from "react";
import classes from "./BestSellers.module.scss";
import ProductList from "../../../../common/components/ProductList/ProductList";
import { Product } from "../../../../types";

const BestSellers: React.FunctionComponent<{ products: Product[] }> = ({
  products,
}) => {
  return (
    <div className={`container ${classes.root}`}>
      <h2>Best sellers</h2>
      <ProductList products={products} />
    </div>
  );
};

export default BestSellers;
