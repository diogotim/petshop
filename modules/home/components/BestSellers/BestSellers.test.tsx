import { render } from "@testing-library/react";
import BestSellers from "./BestSellers";
import mock from "../../../../mock";

describe("BestSellers", () => {
  it("renders the best sellers unchanged", () => {
    const bestSellers = render(<BestSellers products={mock} />);
    expect(bestSellers).toMatchSnapshot();
  });
});
