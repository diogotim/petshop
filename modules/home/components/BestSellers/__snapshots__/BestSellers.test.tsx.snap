// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BestSellers renders the best sellers unchanged 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container root"
      >
        <h2>
          Best sellers
        </h2>
        <div
          class="root"
        >
          <a
            class="prod"
            href="/product/beautiful-scarf"
          >
            <div
              class="productImg"
            >
              <img
                class="imgCover imgFill"
                src="/beautiful-scarf.jpg"
              />
            </div>
            <p
              class="price"
            >
              $
              213.44
            </p>
            <p
              class="prodName"
            >
              Beautiful Scarf
            </p>
          </a>
          <a
            class="prod"
            href="/product/leash-toy"
          >
            <div
              class="productImg"
            >
              <img
                class="imgCover imgFill"
                src="/leash-toy.jpg"
              />
            </div>
            <p
              class="price"
            >
              $
              113.44
            </p>
            <p
              class="prodName"
            >
              Leash Toy
            </p>
          </a>
          <a
            class="prod"
            href="/product/orange-ball"
          >
            <div
              class="productImg"
            >
              <img
                class="imgCover imgFill"
                src="/orange-ball.jpg"
              />
            </div>
            <p
              class="price"
            >
              $
              100
            </p>
            <p
              class="prodName"
            >
              Orange Ball
            </p>
          </a>
          <a
            class="prod"
            href="/product/rope"
          >
            <div
              class="productImg"
            >
              <img
                class="imgCover imgFill"
                src="/rope.jpg"
              />
            </div>
            <p
              class="price"
            >
              $
              98
            </p>
            <p
              class="prodName"
            >
              Rope
            </p>
          </a>
          <a
            class="prod"
            href="/product/seahorse"
          >
            <div
              class="productImg"
            >
              <img
                class="imgCover imgFill"
                src="/seahorse.jpg"
              />
            </div>
            <p
              class="price"
            >
              $
              55.99
            </p>
            <p
              class="prodName"
            >
              Seahorse
            </p>
          </a>
          <a
            class="prod"
            href="/product/small-leash"
          >
            <div
              class="productImg"
            >
              <img
                class="imgCover imgFill"
                src="/small-leash.jpg"
              />
            </div>
            <p
              class="price"
            >
              $
              59.99
            </p>
            <p
              class="prodName"
            >
              Small Leash
            </p>
          </a>
          <a
            class="prod"
            href="/product/super-ball"
          >
            <div
              class="productImg"
            >
              <img
                class="imgCover imgFill"
                src="/super-ball.jpg"
              />
            </div>
            <p
              class="price"
            >
              $
              1930.45
            </p>
            <p
              class="prodName"
            >
              Super Ball
            </p>
          </a>
          <a
            class="prod"
            href="/product/tennis-ball"
          >
            <div
              class="productImg"
            >
              <img
                class="imgCover imgFill"
                src="/tennis-ball.jpg"
              />
            </div>
            <p
              class="price"
            >
              $
              2500
            </p>
            <p
              class="prodName"
            >
              Tennis Ball
            </p>
          </a>
          <a
            class="prod"
            href="/product/yellow-ball"
          >
            <div
              class="productImg"
            >
              <img
                class="imgCover imgFill"
                src="/yellow-ball.jpg"
              />
            </div>
            <p
              class="price"
            >
              $
              12500
            </p>
            <p
              class="prodName"
            >
              Yellow Ball
            </p>
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container root"
    >
      <h2>
        Best sellers
      </h2>
      <div
        class="root"
      >
        <a
          class="prod"
          href="/product/beautiful-scarf"
        >
          <div
            class="productImg"
          >
            <img
              class="imgCover imgFill"
              src="/beautiful-scarf.jpg"
            />
          </div>
          <p
            class="price"
          >
            $
            213.44
          </p>
          <p
            class="prodName"
          >
            Beautiful Scarf
          </p>
        </a>
        <a
          class="prod"
          href="/product/leash-toy"
        >
          <div
            class="productImg"
          >
            <img
              class="imgCover imgFill"
              src="/leash-toy.jpg"
            />
          </div>
          <p
            class="price"
          >
            $
            113.44
          </p>
          <p
            class="prodName"
          >
            Leash Toy
          </p>
        </a>
        <a
          class="prod"
          href="/product/orange-ball"
        >
          <div
            class="productImg"
          >
            <img
              class="imgCover imgFill"
              src="/orange-ball.jpg"
            />
          </div>
          <p
            class="price"
          >
            $
            100
          </p>
          <p
            class="prodName"
          >
            Orange Ball
          </p>
        </a>
        <a
          class="prod"
          href="/product/rope"
        >
          <div
            class="productImg"
          >
            <img
              class="imgCover imgFill"
              src="/rope.jpg"
            />
          </div>
          <p
            class="price"
          >
            $
            98
          </p>
          <p
            class="prodName"
          >
            Rope
          </p>
        </a>
        <a
          class="prod"
          href="/product/seahorse"
        >
          <div
            class="productImg"
          >
            <img
              class="imgCover imgFill"
              src="/seahorse.jpg"
            />
          </div>
          <p
            class="price"
          >
            $
            55.99
          </p>
          <p
            class="prodName"
          >
            Seahorse
          </p>
        </a>
        <a
          class="prod"
          href="/product/small-leash"
        >
          <div
            class="productImg"
          >
            <img
              class="imgCover imgFill"
              src="/small-leash.jpg"
            />
          </div>
          <p
            class="price"
          >
            $
            59.99
          </p>
          <p
            class="prodName"
          >
            Small Leash
          </p>
        </a>
        <a
          class="prod"
          href="/product/super-ball"
        >
          <div
            class="productImg"
          >
            <img
              class="imgCover imgFill"
              src="/super-ball.jpg"
            />
          </div>
          <p
            class="price"
          >
            $
            1930.45
          </p>
          <p
            class="prodName"
          >
            Super Ball
          </p>
        </a>
        <a
          class="prod"
          href="/product/tennis-ball"
        >
          <div
            class="productImg"
          >
            <img
              class="imgCover imgFill"
              src="/tennis-ball.jpg"
            />
          </div>
          <p
            class="price"
          >
            $
            2500
          </p>
          <p
            class="prodName"
          >
            Tennis Ball
          </p>
        </a>
        <a
          class="prod"
          href="/product/yellow-ball"
        >
          <div
            class="productImg"
          >
            <img
              class="imgCover imgFill"
              src="/yellow-ball.jpg"
            />
          </div>
          <p
            class="price"
          >
            $
            12500
          </p>
          <p
            class="prodName"
          >
            Yellow Ball
          </p>
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
